src/main/java/at/hannibal2/skyhanni/data/QuiverAPI.kt

```diff
@@ -0,0 +1,254 @@
+        currentAmount = 0
+        arrowAmount.clear()
+
+        val stacks = event.inventoryItems
+        for (stack in stacks.values) {
+            val lore = stack.getLore()
+            if (lore.isEmpty()) continue
+
+            val arrow = stack.getInternalNameOrNull() ?: continue
+
+            val arrowType = getByNameOrNull(arrow) ?: continue
+            val arrowAmount = stack.stackSize + (this.arrowAmount[arrowType] ?: 0.0f)
+
+            this.arrowAmount[arrowType] = arrowAmount
+        }
+
+        saveArrowAmount()
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    // Inspired by SkyblockFeatures - https://github.com/MrFast-js/SkyblockFeatures/
+    fun onPlaySound(event: PlaySoundEvent) {
+        val holdingBow = InventoryUtils.getItemInHand()?.item is ItemBow && !fakeBows.contains(
+            InventoryUtils.getItemInHand()?.getInternalNameOrNull()
+        )
+
+        // check if sound location is more than configAmount block away from player
+        val soundLocation = event.location
+        if (soundLocation.distanceToPlayer() > SkyHanniMod.feature.dev.bowSoundDistance) return
+
+        if (event.soundName == "random.bow" && holdingBow) {
+            val arrowType = currentArrow ?: return
+            val arrowAmount = QuiverAPI.arrowAmount[arrowType] ?: return
+            if (arrowAmount <= 0) return
+
+            if (InventoryUtils.getChestplate()
+                    ?.getInternalNameOrNull() == "SKELETON_MASTER_CHESTPLATE".asInternalName()
+            ) return
+
+            val infiniteQuiverLevel = InventoryUtils.getItemInHand()?.getEnchantments()?.get("infinite_quiver") ?: 0
+
+            val amountToRemove = {
+                when (Minecraft.getMinecraft().thePlayer.isSneaking) {
+                    true -> 1.0f
+                    false -> {
+                        when (infiniteQuiverLevel) {
+                            in 1..10 -> 1 - (infinityQuiverLevelMultiplier * infiniteQuiverLevel)
+                            else -> 1.0f
+                        }
+                    }
+                }
+            }
+
+            this.arrowAmount[arrowType] = (arrowAmount - amountToRemove()).coerceAtLeast(0.0f)
+
+            saveArrowAmount()
+        }
+    }
+
+    fun hasBowInInvetory(): Boolean {
+        return InventoryUtils.getItemsInOwnInventory().any { it.item is ItemBow }
+    }
+
+    fun getByNameOrNull(name: String): ArrowType? {
+        return arrows.firstOrNull { it.arrow == name }
+    }
+
+    fun getByNameOrNull(internalName: NEUInternalName): ArrowType? {
+        return arrows.firstOrNull { it.internalName == internalName.asString() }
+    }
+
+    fun getByNameOrFlint(name: String): ArrowType {
+        return getByNameOrNull(name) ?: getByNameOrNull("ARROW".asInternalName())!!
+    }
+
+    fun getByNameOrFlint(internalName: NEUInternalName): ArrowType {
+        return getByNameOrNull(internalName) ?: getByNameOrNull("ARROW".asInternalName())!!
+    }
```

Comment:
please rename the functions since there are none descriptive outside of the API or when the only should be used inside the API make them private